Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -7585,20 +7585,71 @@ RefPtr<CSSStyleValue> Parser::parse_grid_shorthand_value(TokenStream<ComponentVa
 RefPtr<CSSStyleValue> Parser::parse_grid_template_areas_value(TokenStream<ComponentValue>& tokens)
 {
     // none | <string>+
-    Vector<Vector<String>> grid_area_rows;
-
     if (auto none = parse_all_as_single_keyword_value(tokens, Keyword::None))
-        return GridTemplateAreaStyleValue::create(move(grid_area_rows));
+        return GridTemplateAreaStyleValue::create({});
+
+    auto is_whitespace = [](u32 code_point) {
```

Comment:
I'm very much not an expert when it comes to CSS, but this is duplicating a bunch of the logic which is in the tokenizer / normally what would be done by TokenStream. E.g `discard_whitespace`

There's probably an ever better way of doing it that I'm not familiar with to layer things better, but would it be better to run the tokenizer on the string? Or nest this somehow?

At least to my reading, it seems to match the spec as well, which states for [\<string\>+](https://www.w3.org/TR/css-values-3/#string-value)

> Tokenize the string into a list of the following tokens, using longest-match semantics:

Reply:
Yeah I wanted to do that but I wasn't sure if I could easily consume a sequence of one or more "." and wasn't sure if it could allow idents starting with non-starting codepoints

Reply:
I am open for suggestions on how to deduplicate this code